### ISSUE TITLE
KafkaJSNotImplemented: LZ4 compression not implemented while there is no compression  configured on producer

### ISSUE THREAD

**Issue by ragoob** [tulios/kafkajs#1141]
**Describe the bug**
I created an automated service that sends a random message every 5 seconds to a topic and the same service consume from this topic and it was working good for some hours and suddenly consumers start crashing "KafkaJSNotImplemented: LZ4 compression not implemented" while I do not provide any compression in producer
**To Reproduce**
Please provide either a link to a:

**Expected behavior**
I expect  no crashing 

**Observed behavior**
A clear and concise description of what *did* happen. Please include any relevant logs [with the log level set to debug](https://kafka.js.org/docs/configuration#logging).

**Environment:**
 - OS: [Kubernetes]
 - KafkaJS version [^1.15.0]
 - Kafka version [2.7.0]
 - NodeJS version [12.13.0]

**Comment by ragoob** [comment]
But I did send a message in compression so why I need to configure this on the consumer

**Comment by tulios** [comment]
Sorry, can you elaborate? You configure the codecs for the entire library, if the consumer is in another machine and the library is not configured with the codecs any message compressed with a codec that is not configured will fail in the consumer. The same with the producer.

**Comment by ragoob** [comment]
let me rephrase, there no configuration for Lz4 in consumer, producer, and the cluster itself. and the messages are not compressed.
in kafkajs documentation the default is "CompressionTypes.None" so I should not get any errors about compression

**Comment by tulios** [comment]
The codec is identified in the message, so it would be hard for the library to attempt to use LZ4 without any information from the message, I would double-check that no messages with LZ4 were published. Feel free to add some logs here as well.

**Comment by ragoob** [comment]
I found the problem is in the topic "compression.type" is LZ4 I do not know why because the topic was with compression anyway it is not a problem in kafkajs so I am sorry for false alarm thanks a lot for your time
